pkg/profilecreator/profilecreator.go

```diff
@@ -281,18 +282,12 @@ func (ghwHandler GHWHandler) GetReservedAndIsolatedCPUs(reservedCPUCount int, sp
 		return "", "", fmt.Errorf("Error determining if Hyperthreading is enabled or not: %v", err)
 	}
 
-	//TODO: Make the allocation logic below more readable by using separate helper functions, one per allocation strategy
-	// (splitReservedCPUsAcrossNUMA=false/true -> two strategies) each one with its clear and nice allocation loop
-	for numaID, node := range topologyInfo.Nodes {
-		if splitReservedCPUsAcrossNUMA {
-			if remainder != 0 {
-				max = (numaID+1)*reservedPerNuma + (numaNodeNum - remainder)
-				remainder--
-			} else {
-				max = max + reservedPerNuma
-			}
+	for numaID, node := range topologyInfoNodes {
+		if remainder != 0 {
+			max = (numaID+1)*reservedPerNuma + (numaNodeNum - remainder)
+			remainder--
 		} else {
-			max = reservedCPUCount
+			max = max + reservedPerNuma
```

Comment:
the only remaining possible improvement is this block (286-291) which we may want to simplify reorganizing the body of the function. The core idea should be:

0. sort cores per-numa node by core id, virtual thread id
1. reserve the first `reservedPerNuma` cores from each numa node 
2. we are left with `remainder` cores to allocate. Iterate over numa node, reserving one core from each up until we reach `remainder`. Right?

Reply:
Well, slightly different. We identify the right number of CPUs that need to be allocated per numa node, meaning reservedPerNuma + (the additional number based on the remainder and the NUMA node)
So best way to explain this is with the help of example here. If the user requests 15  reserved cpus and we have 4 numa nodes. We find reservedPerNuma which in this case is 3 and remainder = 3. So for each numa node we find a max which keeps track of the cumulative resources that should be allocated for each NUMA node:
max = (numaID+1)*reservedPerNuma + (numaNodeNum - remainder) 
For NUMA node 0 max = (0+1)*3 + 4-3 = 4 remainder is decremented => remainder is 2
For NUMA node 1 max = (1+1)*3 + 4-2 = 8 remainder is decremented => remainder is 1
For NUMA node 2 max = (2+1)*3 + 4-2 = 12 remainder is decremented => remainder is 0
For NUMA Node 3 remainder is =0 so 	max = 12 + 3 = 15.

Hope that helps explain things.

Reply:
ok, please copypaste this comment as code comment and as illustrative example and let's move on

Reply:
Ack